### PR TITLE
Fire rate

### PR DESCRIPTION
**Pull request by Ashthecool**
There is now a fire rate to how fast you can shoot every 0.2 milliseconds

### CODE REVIEW

**Comment by tobiasbp** [comment]
You have wrong indentation  starting at 516. After that, you have stuff about game state. That should not be in the game anymore. Right @ivan-BRHM ?

File: my_game.py [hunk 1]
```diff
@@ -628,24 +620,26 @@ def on_key_press(self, key, modifiers):
...
             self.thrust_pressed = True
-        if key == PLAYER_FIRE_KEY:
-            new_shot = PlayerShot(
-                self.player_sprite.center_x,
-                self.player_sprite.center_y
-            )
 
-            self.player_shot_list.append(new_shot)
+        if key == PLAYER_FIRE_KEY:
+            if self.player_shot_fire_rate_counter > 0.2:
```
Comment (tobiasbp) [inline my_game.py:633]: This needs to be a configuration variable

**Review by tobiasbp — CHANGES_REQUESTED** [review]

File: my_game.py [hunk 2]
```diff
@@ -394,7 +394,7 @@ def __init__(self):
 
         # Variable that will hold a list of shots fired by the player
         self.player_shot_list = None
-        self.player_shot_fire_rate_counter = 0
+        self.player_shot_fire_rate_timer = 0
```
Comment (tobiasbp) [inline my_game.py:397]: You need to change the name everywhere it is used.

**Review by tobiasbp — CHANGES_REQUESTED** [review]

File: my_game.py (hunk 2 above)
Comment (Ashthecool) [reply my_game.py:397]: It is done, I've just pushed it Tobias.

File: my_game.py [hunk 3]
```diff
@@ -628,24 +617,25 @@ def on_key_press(self, key, modifiers):
...
-                self.player_sprite.center_x,
-                self.player_sprite.center_y
-            )
 
-            self.player_shot_list.append(new_shot)
-
-        if key == PLAYER_THRUST_KEY:
-            self.player_sprite.thrust()
+        if key == PLAYER_FIRE_KEY:
+            if self.player_shot_fire_rate_timer > PLAYER_FIRE_RATE:
```
Comment (tobiasbp) [inline my_game.py:630]: I guess this should be _>=_, don't you agree?

**Review by tobiasbp — CHANGES_REQUESTED** [review]

**Pull request merged**